### PR TITLE
Cypress tests setup

### LINKED ISSUE #15: Add client-side end-to-end testing
**This ticket is a...**
- [ ] Bug
- [ ] New feature
- [x] Dev QOL change
- [ ] Other?

**Description**
[Cypress](https://www.cypress.io/) is a popular choice for end-to-end testing. It does require some setup though, so this ticket is responsible for bootstrapping the project with Cypress and creating an example test.  The example test can just be a test to ensure the home page loads and the login button renders.

**Acceptance Criteria**
* Cypress tests can be run via npm script
* Sample test exists

### PR DESCRIPTION
**Pull request by windyScripts**
Closes #15 (replace with issue number)
Description

Please include a summary of the changes. Please also include relevant motivation and context. List any dependencies that are required for this change.

Added a test based on cypress for the app being online with a rendered login button.

-     Created a config file in root for cypress
-     Placed cypress tests and related files in ./client/tests/cypress
-     Added a script, cypress:open to open cypress.
-     Added a subsection under the testing section in the readme on running cypress tests

Motivation: To implement front end testing with cypress.

Dependencies: Can be found at [the cypress installation documentation](https://docs.cypress.io/guides/getting-started/installing-cypress)

Testing

Describe what a reviewer should do to test your code, including instructions for manually testing as well as relevant automated test suites to run.

- install dependencies
- start the app
- run cypress:open
- go to e2e from the menu
- run the script cypress-base-test

Type of change

    New feature (non-breaking change which adds functionality)

Screenshots

NA

Checklist:

[x] Changes have new/updated automated tests, if applicable
[x] Changes have new/updated docs, if applicable
[x] I have performed a self-review of my own code
[x] I have added comments on any new, hard-to-understand code
[x] Changes have been manually tested
[x] Changes generate no new errors or warnings

### CODE REVIEW

**Review by timmyichen — COMMENTED** [review]
Looks good to me, tests work locally! One ask is to add some docs so folks know how to run cypress tests locally :)

image

File: client/tests/cypress/e2e/cypress-base-test.ts
Comment (MainlyColors) [inline client/tests/cypress/e2e/cypress-base-test.ts:1]: Fix Formatting, ESLint errors

File: client/tests/cypress/support/e2e.ts
Comment (MainlyColors) [inline client/tests/cypress/support/e2e.ts:1]: Fix Formatting, ESLint error

**Review by MainlyColors — COMMENTED** [review]
Tests Work Locally, just fix those ESLint Errors I mentioned and we're good

image

File: cypress.config.ts
[the remark this answers is not in the document]
Comment (windyScripts) [reply cypress.config.ts:1]: Ran eslint over the branch and dealt with called out errors.

**Comment by DevinCLane** [comment]
Screenshot 2023-03-20 at 12 55 50 PM

works for me here

**Comment by DevinCLane** [comment]
let's add instructions in the docs on how to run these tests

**Comment by MainlyColors** [comment]
Linting is all good now, I agree with Devin & Tim that some docs are required to let others know how to run the cypress tests.

**Comment by timmyichen** [comment]
btw @windyScripts i merged in some docs for serverside testing, so if you pull in `main` you should be able to just add some notes on e2e testing commands in the readme.

File: README.md [hunk 1]
```diff
@@ -93,6 +93,16 @@ You can run in [watch mode](https://jestjs.io/docs/cli#--watch), which is useful
 npm run test:watch -- auth.test.ts
 ```
 
+### UI Testing
+
+We use [cypress](https://docs.cypress.io/guides/overview/why-cypress) for UI testing. Cypress requires the app to be running for testing.
+
+To get started,
+* Execute npm run cypress:open
```
Comment (timmyichen) [inline README.md:101]: Nit: Let's put the command in backticks

```suggestion
* Execute npm `run cypress:open`
```

**Review by timmyichen — APPROVED** [review]
Thanks for adding the docs! one minor non blocking comment

**Pull request merged**